std/typecons.d

```diff
@@ -422,19 +422,31 @@ template Tuple(Specs...)
         U u = U.init;
         T t = u;
     }));
+
     /+ Helper for partial instanciation +/
     template isBuildableFrom(U)
     {
         enum isBuildableFrom(T) = isBuildable!(T, U);
     }
 
+    /+ Determine whether or not all given types in T are the same type +/
+    template areAllSame(T...)
```

Comment:
Do you think there's any value in also supporting Tuples whose types can all implicitly convert to some supertype?